poseidon/controllers/faucet/parser.py

```diff
@@ -86,57 +86,45 @@ def parse_rules(config_file):
         obj_doc = Parser().yaml_in(config_file)
         return obj_doc
 
-    def config(self, config_file, action, port, switch, rules_file=None, endpoints=None, force_apply_rules=None, force_remove_rules=None, coprocess_rules_files=None):
+    def check_mirror(self, config_file, switch, port, obj_doc):
+        if not obj_doc:
+            return False
+        if not self.mirror_ports:
+            self.logger.error('Unable to mirror, no mirror ports defined')
+            return False
+        if 'dps' not in obj_doc:
+            self.logger.warning(f'Unable to find switches in {config_file}')
+            return False
+        else:
```

Comment:
Since all the preceding if's return, can do without the else and pull the next block in left.